drivers/gpio/gpio_mcux_igpio.c

```diff
@@ -44,11 +45,31 @@ static int mcux_igpio_configure(const struct device *dev,
 
 #ifdef CONFIG_PINCTRL
 	struct pinctrl_soc_pin pin_cfg;
+	int cfg_idx = pin;
+
+	/* Some SOCs have non-contiguous gpio pin layouts, account for this */
+	if (IS_ENABLED(CONFIG_SOC_MIMXRT1015) &&
+		(config->port_num == 3) && (pin >= 20)) {
+		/* RT1015 does not have GPIO3 pins 4-19 */
+		cfg_idx -= 16;
+	} else if (IS_ENABLED(CONFIG_SOC_MIMXRT1015) &&
+		(config->port_num == 5)) {
```

Comment:
how about rt1010?

Reply:
For GPIO5? Those pins seem to be contiguous: 
![image](https://user-images.githubusercontent.com/34665051/163412151-50771d3e-bb64-4d9d-97cc-8ac75fffbcf5.png)

If you mean the gap in GPIO3 pins, that is handled in the next `else if` case

Reply:
I did not see the RT1010 in hte else if only 1024 and 1021